src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/rc_command_request.cc

```diff
@@ -0,0 +1,282 @@
+      return false;
+    }
+    case AcquireResult::ASK_DRIVER: {
+      SetResourceState(module_type, ResourceState::BUSY);
+      SendGetUserConsent(module_type);
+      return false;
+    }
+    case AcquireResult::REJECTED: {
+      SetResourceState(ModuleType(), ResourceState::FREE);
+      SendResponse(false, mobile_apis::Result::REJECTED, "");
+      return false;
+    }
+  }
+  return false;
+}
+
+void RCCommandRequest::on_event(const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (event.id() == hmi_apis::FunctionID::RC_GetInteriorVehicleDataConsent) {
+    ProcessAccessResponse(event);
+  } else {
+    const std::string module_type = ModuleType();
+    SetResourceState(module_type, ResourceState::FREE);
+  }
+}
+
+void RCCommandRequest::ProcessAccessResponse(
+    const app_mngr::event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  app_mngr::ApplicationSharedPtr app =
+      application_manager_.application(CommandRequestImpl::connection_key());
+  const std::string module_type = ModuleType();
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "NULL pointer.");
+    SetResourceState(ModuleType(), ResourceState::FREE);
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED, "");
+    return;
+  }
+
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  mobile_apis::Result::eType result_code =
+      GetMobileResultCode(static_cast<hmi_apis::Common_Result::eType>(
+          message[app_mngr::strings::params][app_mngr::hmi_response::code]
+              .asUInt()));
+
+  const bool result =
+      helpers::Compare<mobile_apis::Result::eType, helpers::EQ, helpers::ONE>(
+          result_code,
+          mobile_apis::Result::SUCCESS,
+          mobile_apis::Result::WARNINGS);
+
+  bool is_allowed = false;
+  if (result) {
+    if (message[app_mngr::strings::msg_params].keyExists(
+            message_params::kAllowed)) {
+      is_allowed =
+          message[app_mngr::strings::msg_params][message_params::kAllowed]
+              .asBool();
+    }
+    if (is_allowed) {
+      resource_allocation_manager_.ForceAcquireResource(module_type,
+                                                        app->app_id());
+      Execute();  // run child's logic
+    } else {
+      resource_allocation_manager_.OnDriverDisallowed(module_type,
+                                                      app->app_id());
+      SetResourceState(ModuleType(), ResourceState::FREE);
+      SendResponse(
+          false,
+          mobile_apis::Result::REJECTED,
+          "The resource is in use and the driver disallows this remote "
+          "control RPC");
+    }
+  } else {
+    std::string response_info;
+    GetInfo(message, response_info);
+    SetResourceState(ModuleType(), ResourceState::FREE);
+    SendResponse(false, result_code, response_info.c_str());
+  }
+}
+
+void RCCommandRequest::SendGetUserConsent(const std::string& module_type) {
```

Comment:
still don't know how to get 20s timeout for the request (10s default + 10s for GetUserConsent)